tidy3d/components/data/dataset.py

```diff
@@ -73,6 +74,18 @@ def colocate(self, x=None, y=None, z=None) -> xr.Dataset:
         Be sure to apply this method to your field data in those cases.
         """
 
+        try:
```

Comment:
If we have to do it this way, I'd think that `if hasatttr(self, "monitor"):` is the lesser of two evils.

Reply:
The other thing I started doing first was to overwrite the method in MonitorData, but then I have to copy the doc string and it felt kinda ugly too

Reply:
Updated.